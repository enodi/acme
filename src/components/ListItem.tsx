import React, { useState } from "react";

import { ReactComponent as Star } from "../assets/star.svg";

const ListItem: React.FC<{
  toggleConversationType: any;
  item: any;
}> = ({ toggleConversationType, item }) => {
  const [isSelected, setIsSelected] = useState(false);

  const toggleMessageField = () => {
    setIsSelected(!isSelected);
    localStorage.setItem(
      "selectedId",
      JSON.stringify({ id: item.id, selected: true })
    );
  };

  return (
    <div
      className={`list-item ${
        isSelected ? "list-item--selected" : "list-item--not-selected"
      }`}
      onClick={toggleMessageField}
    >
      <div className="list-item__profile-img" />
      <div className="list-item__user-section">
        <h4 className="list-item__name">{`${item.firstName} ${item.lastName}`}</h4>
        <p className="list-item__message">Lorem Ipsum</p>
      </div>
      <div className="list-item__favourites-section">
        <p className="list-item__time-sent">1h</p>
        <Star
          className="list-item__star"
          fill={item.isFavorite ? "#333333" : "#d2d2d2"}
          onClick={(e) => toggleConversationType(e, item.id)}
        />
      </div>
    </div>
  );
};

export default ListItem;
