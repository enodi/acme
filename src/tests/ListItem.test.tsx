import React from "react";
import { shallow } from "enzyme";

import ListItem from "../components/ListItem";
import { selectedItem } from "./fixtures";

let toggleConversationType, handleSelected, wrapper: any;
beforeEach(() => {
  toggleConversationType = jest.fn();
  handleSelected = jest.fn();
  wrapper = shallow(
    <ListItem
      toggleConversationType={toggleConversationType}
      item={selectedItem}
      handleSelected={handleSelected}
    />
  );
});

test("should render ListItem component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should find text rendered on list item", () => {
  expect(wrapper.find("h4").text()).toEqual("Kirby Rotte");
});
