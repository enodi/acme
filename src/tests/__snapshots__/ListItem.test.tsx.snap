// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ListItem component correctly 1`] = `
<div
  className="list-item list-item--selected"
  onClick={[Function]}
>
  <div
    className="list-item__profile-img"
  />
  <div
    className="list-item__user-section"
  >
    <h4
      className="list-item__name"
    >
      Kirby Rotte
    </h4>
    <p
      className="list-item__message"
    >
      Lorem Ipsum
    </p>
  </div>
  <div
    className="list-item__favourites-section"
  >
    <p
      className="list-item__time-sent"
    >
      1h
    </p>
    <ForwardRef(SvgStar)
      className="list-item__star"
      fill="#333333"
      onClick={[Function]}
    />
  </div>
</div>
`;
